### PR TITLE
feat: enhance increment and decrement functionality number component

### PR DESCRIPTION
**Pull request by brampeirs**
Notes:
- Added logic to `onIncrement` and `onDecrement` to constrain values within min and max bounds
- Updated tests to verify min/max constraints
- Conform with React implementation

### Increment
Value | Step | Min | Max | Result before | Result after
-- | -- | -- | -- | -- | --
0 | 10 | 15 |   | 10 | 15
95 | 10 |   | 100 | 95 | 100
105 | 10 |   | 100 | 105 | 100

### Decrement

Value | Step | Min | Max | Result before | Result after
-- | -- | -- | -- | -- | --
20 | 2 |   | 15 | 18 | 15
6 | 2 | 5 |   | 6 | 5
10 | 2 | 15 |   | 10 | 15

https://github.com/user-attachments/assets/dc16654b-c70e-4071-ad19-e6f9de0efbb6

### CODE REVIEW

File: src/number-input/number.component.spec.ts [hunk 1]
```diff
@@ -129,6 +152,29 @@ describe("Number", () => {
 		expect(component.value).toEqual(0);
 	});
 
+	it("should decrement and set value to max if value - step is less than max", () => {
+		fixture.detectChanges();
+		buttonUp = fixture.debugElement.query(By.css(".up-icon")).nativeElement;
```
Comment (Akshat55) [inline src/number-input/number.component.spec.ts:157]: Should this be `.down-icon`? 🤔

**Review by Akshat55 — CHANGES_REQUESTED** [review]
LGTM! Just a question about one of the tests.

File: src/number-input/number.component.spec.ts (hunk 1 above)
Comment (brampeirs) [reply src/number-input/number.component.spec.ts:157]: Good catch, I'll update the test. Oversight from my side.

Comment (brampeirs) [reply src/number-input/number.component.spec.ts]: fixed, could you review once more?

**Review by Akshat55 — APPROVED** [review]

**Review by zvonimirfras — APPROVED** [review]

**Pull request merged**